### DISCUSSION TITLE
Are there any request body extractors that implement std::io::Read?

### DISCUSSION THREAD

**Discussion by phish3y** [tokio-rs/axum#2084] [Q&A] [answered]
I can't seem to identify any https://docs.rs/axum/latest/axum/extract/index.html#request-body-extractors that implement https://doc.rust-lang.org/stable/std/io/trait.Read.html. I am wondering if I'm missing something? If in fact none of them implement std::io::Read I would love to understand why?

I'm trying to pass a request body to the tar crate directly in memory. I know I can do it via a file read/write, and I know I can do it in memory using std::io::Cursor.

Thanks!

**Comment by davidpdrsn** [comment, marked as answer]
Nope. Bodies are async streams whereas std::io::Read is sync.
